### ISSUE TITLE
Failed to decrypt password --- This needs to be run as the user who owns the password

### ISSUE THREAD

**Issue by LawnSounds** [ash47/EnterpriseWifiPasswordRecover#1]
Getting this error when run as system (PsExec.exe), administrator CMD and normal user CMD I'm getting the error "Run as SYSTEM or ORIGINAL USER!". 

_me stupid solved sorry_

**Comment by Benji-Collins** [comment]
You gonna say how you solved it?

**Comment by ash47** [comment]
Are you facing the same issue? If you can provide some proper details i can
probably help

**Comment by Benji-Collins** [comment]
I'm trying to get my own password that I used to connect to my school's wifi. Each student has a number and their own password. When I run the program on my laptop (with admin) I get this:

`C:\WINDOWS\system32>C:\Users\colli\Downloads\EnterpriseWifiPasswordRecover.exe`

`Extracted stage1 for {7D2313AC-E665-4021-AA27-B6D98BD70715}`

`Found encrypted password blob...`

`Failed to decrypt password --- This needs to be run as the user who owns the password`

`Found the following:`

`Domain: oakbank`

`Username: 150015`

`Password:`

Is the user that owns the password the 150015 user? Because that is a domain account. I log in to my laptop with my own local account. I would have thought that the tool could get the password anyway, as long as it was entered on the device.

**Comment by ash47** [comment]
It should be like this:
 - use psexec to run the tool as system (you seemed to do this correctly)
 - run the tool as the user that you are logged in at the time you
connected to the WiFi, this would be your local account

**Comment by Benji-Collins** [comment]
How exactly do I run the program again as the standard user? I tried to run the exe in a new administrator terminal window but I got a “This app can’t run on your PC” warning. Upon clicking ok the program (in cmd) closed with “Access denied”. When I double click the exe like a regular program is creates a folder called profiles. The only file inside is a bunch of numbers with random characters inside it. I haven’t tried to open that txt with anything other than notepad yet.

**Comment by ash47** [comment]
Did you perform the psexec step successfully?

**Comment by Benji-Collins** [comment]
Well it outputs what I posted above. That step doesn’t create any extra files/folders. Would be great if you could provide screenshots of what a successful run looks like. If I don’t know what the output should be, I don’t know if it’s worked :)

**Comment by ash47** [comment]
Will you please do:

 - Start an admin cmd
 - run: psexec -s -i cmd
 A new cmd should appear, go into that cmd
 - run: whoami
Tell me the result
 - use the "cd <directory>" command to go to the folder with the tool
 - run the tool

This should produce some files

- change back to the admin cmd
 - go to the same folder using cmd
 - run the command

Your creds should be printed to the screen

**Comment by Benji-Collins** [comment]
Sure.
- Started cmd as administrator
- Ran `<pathtofile>\PsExec.exe -s -i cmd` and a new cmd window came up
- `whoami` returns `nt authority\system`
- Ran `cd <pathtofile>`
- Ran `EnterpriseWifiPasswordRecover.exe`
I get [this warning](https://i.imgur.com/Y8b83JB.png). When I click *Close* the cmd windows says `Access denied`. I am currently on Windows 10 1803 (Build 17134.191). The same thing happens when I set the tool to run in Windows 8 or 7 compatibility mode.

**Comment by Benji-Collins** [comment]
Any ideas?

**Comment by ash47** [comment]
Are you running some weird version of windows? The app is written in C#,
looks like i wrote it it in .net version 2.0, which is very old -- perhaps
confirm that you have .NET version 2.0 installed?

Possibly this?

https://www.microsoft.com/en-AU/download/details.aspx?id=6523

That's the only reason I can think that it would tell you that app cant run
on your PC.

**Comment by Benji-Collins** [comment]
I downloaded that file but it didn't run for some reason so I checked out the *Turn Windows features on or off* part of Control Panel and found that .Net 2.0 is indeed pre-installed. I disabled it and restarted my PC. When I ran that .Net 2.0 installer again, I was prompted to install the .Net 3.5 (with 2.0) package. I clicked yes and it was installed. Then I went through your instructions again and the tool worked flawlessly. Thank you for that. 

Still not sure why it didn't work straight away though. Must have just been some weird Windows bug.
